Sasquatch/Sasquatch/ViewControllers/Sections/CommonSchemaPropertiesTableSection.swift

```diff
@@ -1,94 +1,99 @@
   var transmissionTargetSelectorCell: MSAnalyticsTransmissionTargetSelectorViewCell?
-  
-  enum CommonSchemaProperty : Int {
-    case AppName = 1
-    case AppVersion
-    case AppLocale
-  }
-  
-  var propertyTuples = [("appName", ""), ("appVersion", ""), ("appLocale", "")]
-  
+  var collectDeviceIdStates: [String: Bool]!
+
   override init(tableSection: Int, tableView: UITableView) {
     super.init(tableSection: tableSection, tableView: tableView)
-    targets = [String: [(String, String)]]()
+    propertyValues = [String: [String]]()
+    collectDeviceIdStates = [String: Bool]()
     let appName = Bundle.main.infoDictionary![kCFBundleNameKey as String] as! String
     let parentTargetToken = appName == "SasquatchSwift" ? kMSSwiftRuntimeTargetToken : kMSObjCRuntimeTargetToken
-    targets[parentTargetToken] = [(String, String)](propertyTuples)
-    targets[kMSTargetToken1] = [(String, String)](propertyTuples)
-    targets[kMSTargetToken2] = [(String, String)](propertyTuples)
+    for token in [parentTargetToken, kMSTargetToken1, kMSTargetToken2] {
+      propertyValues[token] = Array(repeating: "", count: propertyKeys.count)
+      collectDeviceIdStates[token] = false
+    }
     transmissionTargetSelectorCell = loadCellFromNib()
     transmissionTargetSelectorCell?.didSelectTransmissionTarget = reloadSection
   }
   
   override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-    if indexPath.row == 0 {
+    switch indexPath.row {
+    case kTargetSelectorCellRow:
       return transmissionTargetSelectorCell!
-    } else {
-      let cell: MSAnalyticsPropertyTableViewCell? = loadCellFromNib()
-      cell?.valueField.placeholder = "Override value"
-      
-      // Set cell text.
-      var label : String = ""
-      switch indexPath.row {
-      case CommonSchemaProperty.AppName.rawValue:
-        label = "App Name"
-        break
-        
-      case CommonSchemaProperty.AppVersion.rawValue:
-        label = "App Version"
-        break
-        
-      case CommonSchemaProperty.AppLocale.rawValue:
-        label = "App Locale"
-        break
-      default:
-        break
-      }
-      cell!.keyField.text = label
-      
-      let selectedTarget = transmissionTargetSelectorCell?.selectedTransmissionTarget()
-
-      cell!.valueField.text = targets[selectedTarget!]![indexPath.row - 1].1
-      
-      // Set cell to respond to being edited.
-      cell!.valueField.addTarget(self, action: #selector(propertyValueChanged), for: .editingChanged)
-      cell!.valueField.addTarget(self, action: #selector(dismissKeyboard), for: .editingDidEndOnExit)
-      return cell!
+    case kDeviceIdRow:
+        let cell = tableView.dequeueReusableCell(withIdentifier: switchCellIdentifier)!
+        let switcher: UISwitch? = cell.getSubview()
+        let selectedTarget = transmissionTargetSelectorCell?.selectedTransmissionTarget()!
+        switcher!.isOn = collectDeviceIdStates[selectedTarget!]!
```

Comment:
` Fatal error: Unexpectedly found nil while unwrapping an Optional value`

Reply:
I fixed the crash and cleaned up the underlying implementation to fix the runtime target tokens for Sasqutch ObjC and Puppet.